### PR TITLE
Docs: fix broken links

### PR DESCRIPTION
**Pull request by felixxm**
----
📚 Documentation preview 📚: https://cpython-previews--116651.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/pathlib.rst [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 .. versionadded:: 3.4
 
-**Source code:** :source:`Lib/pathlib.py`
+**Source code:** :source:`Lib/pathlib/`
```
Comment (felixxm) [inline Doc/library/pathlib.rst:10]: This cannot be backported. I can move it to a separate commit or manually prepare PRs for backporting.

**Review by erlend-aasland — APPROVED** [review]
Thanks!

File: Doc/library/pathlib.rst (hunk 1 above)
Comment (erlend-aasland) [reply Doc/library/pathlib.rst:10]: It's ok to fix it in a manual backport.

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
@felixxm would you be able to do the manual 3.11 backport?

**Comment by felixxm** [comment]
> @felixxm would you be able to do the manual 3.11 backport?

Sure, #116701.

**Pull request merged**